java/sdk/java/src/main/java/dev/feast/FeastClient.java

```diff
@@ -118,11 +117,56 @@ public GetFeastServingInfoResponse getFeastServingInfo() {
    * @param featureRefs list of string feature references to retrieve in the following format
    *     featureTable:feature, where 'featureTable' and 'feature' refer to the FeatureTable and
    *     Feature names respectively. Only the Feature name is required.
-   * @param rows list of {@link Row} to select the entities to retrieve the features for.
+   * @param entities list of {@link Row} to select the entities to retrieve the features for.
    * @return list of {@link Row} containing retrieved data fields.
    */
-  public List<Row> getOnlineFeatures(List<String> featureRefs, List<Row> rows) {
-    return getOnlineFeatures(featureRefs, rows, "");
+  public List<Row> getOnlineFeatures(List<String> featureRefs, List<Row> entities) {
+    GetOnlineFeaturesRequest.Builder requestBuilder = GetOnlineFeaturesRequest.newBuilder();
+
+    requestBuilder.setFeatures(
+        ServingAPIProto.FeatureList.newBuilder().addAllVal(featureRefs).build());
+
+    Map<String, ValueProto.RepeatedValue.Builder> columnarEntities = new HashMap<>();
```

Comment:
maybe abstract 129-137 out into a getEntityValuesMap() method? and then you can put all the entities at once